File: f77/build02/diffeq.f
C
      Subroutine Differential_Equations
C     This routine evaluates the state derivatives for the
C     mass-spring-damper system.
C
C     Inputs:
C       Damping_Coefficient - Damping force per velo'city [N/m/s]
C       Gravity             - Acceleration due to gravity [m/sec**2]
C       Mass                - Mass suspended from spring [Kg]
C       Spring_Coefficient  - Restoring force per position [N/m]
C       X                   - Position of suspended mass (m]
C       Xd                  - Velocity of suspended mass [m/sec]
C
C     Outputs:
C       Xdd - Acceleration of suspended mass [m/sec**2]
C
C     Internal variables and constants:
C       None.
C
C     Declare global common
C
      Include 'global.inc'
      Include 'sysvars.inc'
      Equivalence (Real_Array(10), Damping_Coefficient)
      Equivalence (Real_Array(11), Gravity)
      Equivalence (Real_Array(12), Mass)
      Equivalence (Real_Array(13), Spring_Coefficient)
      Equivalence (Real_Array(16), X)
      Equivalence (Real_Array(17), Xd)
      Equivalence (Real_Array(18), Xdd)
C
C  Print states for this time
C
 100     Format(' ',3ES15.5)
         Write(*,100) Time, X, Xd
C
C       Calculate derivative at current time
C
      Xdd = -(Spring_Coefficient*X + Damping_Coefficient*Xd)/Mass -
     &     Gravity
C
      Return
      End

File: compcert/build02/diffeq.c
#include <stdio.h>
#include "global.h"
#include "sysvars.h"
void differential_equations() {
  #define damping_coefficient real_array[10].r  // Damping force per velo'city [N/m/s]
  #define gravity             real_array[11].r  // Acceleration due to gravity [m/sec**2]
  #define mass                real_array[12].r  // Restoring force per position [N/m]
  #define spring_coefficient  real_array[13].r  // Mass suspended from spring [Kg]
  #define x                   real_array[16].r  // Position of suspended mass (m]
  #define xd                  real_array[17].r  // Velocity of suspended mass [m/sec]
  #define xdd                 real_array[18].r  // Acceleration of suspended mass [m/sec**2]
  // Print states for this time
  printf("%.4f\t%.6f\t%.6f\n",time, x, xd);
  // Calculate derivative at current time
  xdd = -(spring_coefficient*x+damping_coefficient*xd)/mass - gravity;
}
